app/models/openscap_result.rb

```diff
@@ -0,0 +1,77 @@
+class OpenscapResult < ApplicationRecord
+  belongs_to :container_image
+  has_one    :binary_blob,           :dependent => :destroy, :autosave => true, :as => :resource
+  has_many   :openscap_rule_results, :dependent => :destroy, :autosave => true
+
+  def self.openscap_available?
+    # needed only for travis
+    require 'openscap'
+    require 'openscap/ds/arf'
+    require 'openscap/xccdf/benchmark'
+    true
+  rescue LoadError
+    false
+  end
+
+  def html
+    raw = binary_blob.try(:binary)
+    return unless raw
+    with_openscap_arf(raw) do |arf|
+      ascii8bit_to_utf8(arf.html)
+    end
+  end
+
+  def raw=(value)
+    self.binary_blob = BinaryBlob.new(:name => 'openscap_compliance_arf', :data_type => 'XML')
+    binary_blob.binary = value
+    with_openscap_objects(binary_blob.binary) do |rule_results, benchmark_items|
+      create_results(rule_results, benchmark_items)
```

Comment:
That is right @matthewd 
Per image I always:
- Destroy the old openscap_result (and openscap_rule_results due to :autosave => true)
- Create a new one
- Set raw once

Looks like the example above would leave dangling openscap_rule_results. I added a check at raw= to make sure that does not happen
```
raise "Cannot invoke raw setter more than once." unless binary_blob.nil?
```